src/box/iproto.cc

```diff
@@ -3431,3 +3589,56 @@ iproto_thread_rmean_foreach(int thread_id, void *cb, void *cb_ctx)
 		rc = 1;
 	return rc;
 }
+
+int
+iproto_override(uint32_t rq_type, iproto_rq_handler_f cb,
+		iproto_rq_handler_dtor_f dtor, void *ctx)
+{
+	if (rq_type == IPROTO_JOIN || rq_type == IPROTO_FETCH_SNAPSHOT ||
+	    rq_type == IPROTO_REGISTER || rq_type == IPROTO_SUBSCRIBE) {
+		const char *feature = tt_sprintf("%s request type",
+						 iproto_type_name(rq_type));
+		diag_set(ClientError, ER_UNSUPPORTED,
+			 "IPROTO request handler overriding", feature);
+		return -1;
+	}
+
+	if (cb == NULL) {
+		mh_int_t k = mh_i32ptr_find(tx_rq_handlers, rq_type, NULL);
+		if (k != mh_end(tx_rq_handlers)) {
+			struct mh_i32ptr_node_t *node =
+				mh_i32ptr_node(tx_rq_handlers, k);
+			struct iproto_rq_handler *old =
+				(struct iproto_rq_handler *)node->val;
+			if (old->dtor != NULL)
+				old->dtor(old->ctx);
+			free(old);
+			mh_i32ptr_del(tx_rq_handlers, k, NULL);
+			iproto_cfg_override(rq_type, false);
```

Comment:
I think that the handler should be removed from the iproto threads first and only then from the tx thread. Otherwise, a request may get sent via the override route after the handler was removed.

Reply:
Yes, this is apparently a bug, fixed.